packages/ifpack2/src/Ifpack2_LocalFilter_def.hpp

```diff
@@ -679,6 +682,11 @@ apply (const Tpetra::MultiVector<scalar_type,local_ordinal_type,global_ordinal_t
   }
 #endif // HAVE_IFPACK2_DEBUG
 
+  TEUCHOS_TEST_FOR_EXCEPTION(
+    getBlockSize() > 1, std::runtime_error,
+    "Ifpack2::LocalFilter::apply: Block size greater than zero is not yet supported for "
+    "LocalFilter::apply. Please contact an Ifpack2 developer to request this feature.");
+
```

Comment:
I'm assuming by `Ifpack2 interfaces` you mean the preconditioner factory? I have just gone in and added MDF as an option here. Otherwise one could also construct an `Ifpack2::MDF` directly. Happy to make more changes if there is some other interface I'm missing here.

As for `LocalFilter::apply`, that is correct that `AdditiveSchwarz` doesn't actually call `LocalFilter::apply` but rather just uses that matrix in its own `localApply` call. MDF in general doesn't really tie into the local filter, except that the main use case is using some scalar matrix to generate a reordering to use with AdditiveSchwarz for other matrices.